utils/retry/retry.go

```diff
@@ -0,0 +1,60 @@
+package retry
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// BackoffExponential makes maxRetry attempts to executed function f. After the nth attempt,
+// we wait retryMilliseconds*2^n ms before retrying. Returns an error after
+// maxRetry unsuccessful attempts. It will short circuit if ctx is Done.
+func BackoffExponential(ctx context.Context, retryFuncName string, f func(context.Context) error, maxRetry int, retryMilliseconds int, logger zerolog.Logger) bool {
+	wait := time.Duration(retryMilliseconds) * time.Millisecond
+	for attempt := 1; attempt <= maxRetry; attempt++ {
+		err := f(ctx)
+		if err != nil {
+			logger.Warn().Err(err).Str("retrying_func", retryFuncName).Msgf("attempt %d/%d failed", attempt, maxRetry)
```

Comment:
It would be nice to include the time elapsed and retry attempt number as structured fields, similar to https://github.com/onflow/flow-go/blob/6cb1baeb5f0a6177de7a41a2501d46edb3c7ca8a/module/epochs/base_client.go#L84